pulp_2to3_migration/tests/functional/test_rpm_repo.py

```diff
@@ -0,0 +1,215 @@
+            with self.subTest(repo=repo):
+                version_count = 2 if PULP_2_RPM_DATA['content'][repo.name] else 1
+                self.assertEqual(
+                    self.rpm_repo_versions_api.list(repo.pulp_href).count, version_count
+                )
+                # content count per repo
+                for content_type, api in self.rpm_content_apis.items():
+                    with self.subTest(content_type=content_type):
+                        repo_content = api.list(repository_version=repo.latest_version_href)
+                        self.assertEqual(
+                            repo_content.count,
+                            PULP_2_RPM_DATA['content'][repo.name].get(content_type, 0)
+                        )
+
+    def test_rpm_importer_migration(self):
+        """
+        Test that RPM importers are correctly migrated.
+        """
+        self.assertEqual(self.rpm_remote_api.list().count, PULP_2_RPM_DATA['remotes'])
+        for remote in self.rpm_remote_api.list().results:
+            with self.subTest(remote=remote):
+                repo_name = '-'.join(remote.name.split('-')[1:])
+                repo_url = os.path.join(FIXTURES_BASE_URL, repo_name) + '/'
+                self.assertEqual(remote.url, repo_url)
+                self.assertEqual(remote.policy, 'on_demand')
+
+    def test_rpm_distributor_migration(self):
+        """
+        Test that RPM distributors are correctly migrated.
+        """
+        self.assertEqual(self.rpm_publication_api.list().count, PULP_2_RPM_DATA['publications'])
+        self.assertEqual(self.rpm_distribution_api.list().count, PULP_2_RPM_DATA['distributions'])
+        for dist in self.rpm_distribution_api.list().results:
+            with self.subTest(dist=dist):
+                base_path = '-'.join(dist.name.split('-')[1:])
+                self.assertEqual(dist.base_path, base_path)
+
+
+class SimplePlanMixin:
+    @classmethod
+    def run_migration(cls):
+        mp = cls.migration_plans_api.create({'plan': RPM_SIMPLE_PLAN})
+        mp_run_response = cls.migration_plans_api.run(mp.pulp_href, {})
+        task = monitor_task(mp_run_response.task)
+        monitor_task_group(task.task_group)
+
+
+class ComplexPlanMixin:
+    @classmethod
+    def run_migration(cls):
+        mp = cls.migration_plans_api.create({'plan': RPM_COMPLEX_PLAN})
+        mp_run_response = cls.migration_plans_api.run(mp.pulp_href, {})
+        task = monitor_task(mp_run_response.task)
+        monitor_task_group(task.task_group)
+
+
+@unittest.skip('empty repos are not migrated until https://pulp.plan.io/issues/6516 is done')
+class TestRpmRepoMigrationSimplePlan(BaseTestRpmRepo, unittest.TestCase, SimplePlanMixin):
+    """
+    Test RPM repo migration using simple migration plan.
+    """
+
+
+class TestRpmRepoMigrationComplexPlan(BaseTestRpmRepo, unittest.TestCase, ComplexPlanMixin):
+    """
+    Test RPM repo migration using complex migration plan.
+    """
```

Comment:
Tell me if it's an overcomplication or not.
I'm open for suggestion on how to run the same tests for different plans.

I'd like to decide on one way and then use it everywhere.